File: apps/docs/content/guides/auth/examples/managing-user-data/addingAndRetrievingUserData.kt
val data = supabase.auth.signUpWith(Email) {
    email = "valid.email@supabase.io"
    password = "example-password"
    data = buildJsonObject {
        put("first_name", "John")
        put("age", 27)
    }
}

val user = supabase.auth.retrieveUserForCurrentSession()
//Or you can use the user from the current session:
val user = supabase.auth.currentUserOrNull()
val metadata = user?.userMetadata

File: apps/docs/content/guides/auth/examples/managing-user-data/addingAndRetrievingUserData.js
const { data, error } = await supabase.auth.signUp({
    email: 'valid.email@supabase.io',
    password: 'example-password',
    options: {
        data: {
            first_name: 'John',
            age: 27,
        },
    },
})

const {
    data: { user },
} = await supabase.auth.getUser()
let metadata = user?.user_metadata